### ISSUE TITLE
Mapper.RegisterCustom is not used when source value is default<T>

### ISSUE THREAD

**Issue by high1** [fluentsprings/ExpressMapper#103]
I wrote about this here:
https://github.com/fluentsprings/ExpressMapper/issues/96, and I'm sure that this is an issue in Mapper.RegisterCustom. Basically, registered custom mapping is not used when source value is default<T> - Mapper.RegisterCustom<int, string>(src => src.ToString("#0.0"); will not work if src is 0, but all other values are fine. This is an ad-hoc unit test:

```
[Test]
        public void DefaultValueShouldAlsoUseRegisterCustom()
        {
            Mapper.Register<TestDefaultDecimal, TestDefaultString>()
                .Member(dest => dest.TestString, src => src.TestDecimal);
            Mapper.RegisterCustom<decimal, string>(src => src.ToString("#0.00"));
            Mapper.Compile();
            var test = new TestDefaultDecimal() { TestDecimal = 1 };
            var result = Mapper.Map<TestDefaultDecimal, TestDefaultString>(test);
            Assert.AreEqual(result.TestString, "1.00");
            test.TestDecimal = default(decimal);
            result = Mapper.Map<TestDefaultDecimal, TestDefaultString>(test);
            //This is where the mapping fails
            Assert.AreEqual(result.TestString, "0.00");
        }

        public class TestDefaultDecimal
        {
            public decimal TestDecimal { get; set; }
        }

        public class TestDefaultString
        {
            public string TestString { get; set; }
        }
```
The issue is in:
MappingServiceBase.cs in function GetMemberMappingExpression:
```
var checkNullExp =
                        Expression.IfThenElse(Expression.Equal(right, Expression.Default(right.Type)),
                            Expression.Assign(left, Expression.Default(left.Type)), resultBlockExp);
```
This code assigns default values. Not sure if this is by design, but I think that it should not be applied to primitive types and some other exceptions - i.e. string... Maybe
`t.IsPrimitive || t.IsValueType || (t == typeof(string));`

**Comment by anisimovyuriy** [comment]
Thanks a lot @high1 for your great input and fix right away!

**Comment by anisimovyuriy** [comment]
I'm fixing it right now. Just a workaround: Don't use ```Register.Custom```, use the following:

```
Mapper.Register<TestDefaultDecimal, TestDefaultString>()
                .Member(dest => dest.TestString, src => src.TestDecimal.ToString("#0.00"));
```

**Comment by anisimovyuriy** [comment]
Fixed! Will be included in 1.8.5.
Thanks a lot @high1!
